api/v1/testkube.yaml

```diff
@@ -2791,6 +3009,102 @@ components:
               type: string
             example: ["logline1", "logline2", "logline3"]
 
+    TestTrigger:
+      type: object
+      required:
+        - resource
+        - resourceSelector
+        - event
+        - execution
+        - testSelector
+      properties:
+        name:
+          type: string
+          description: test trigger name
+          example: "test1"
+        namespace:
+          type: string
+          description: test trigger namespace
+          example: "testkube"
+        labels:
+          type: object
+          description: "test trigger labels"
+          additionalProperties:
+            type: string
+          example:
+            env: "prod"
+            app: "backend"
+        resource:
+          type: string
+          description: Kubernetes resource for which we are creating trigger
+          example: pod
+        resourceSelector:
+          $ref: "#/components/schemas/TestTriggerSelector"
+        event:
+          type: string
+          description: listen for event for selected resource
+          example: modified
+        action:
+          type: string
+          description: action to perform when an event matches this trigger
+          example: run
+        execution:
```

Comment:
execution is a result ot test/test suite run, might be object or actionResource

Reply:
Hmm you make a good point, will need to discuss with Michael and Bruno again for the TestTrigger field names